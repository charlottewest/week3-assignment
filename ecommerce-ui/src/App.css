div.container {
  padding: 80px;
}

h1.airbnbSectionHeader {
  font-size: 20px;
  text-decoration: underline;
}

div.shoppingCartContainer {
  padding: 50px;
}

div.availablePropertiesContainer {
  padding: 50px;
}
