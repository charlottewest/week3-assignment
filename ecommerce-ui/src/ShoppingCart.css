div.shoppingCart {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 10px;
  grid-template-columns: repeat(2, 1fr);
  padding: 50px;
}

div.totalPayment {
  grid-column: 2;
  grid-row: 2;
  justify-self: end;
}
