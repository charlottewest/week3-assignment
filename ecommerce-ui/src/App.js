import React from 'react';
import PropTypes from 'prop-types';
import ShoppingCart from './ShoppingCart';
import Airbnb from './Airbnb';
import AirbnbList from './AirbnbList';
import Airbnbs from './airbnbs.json';
import './App.css';
import './Airbnb.css';

class App extends React.Component {
  constructor(props){
    super(props);
    this.state = {
      airbnbs: Airbnbs,
      selectedAirbnbs: [],
      totalPayment: 0
    }
  }

  bookAirbnb = (index) => {
    const selectedAirbnb = this.state.airbnbs[index];
    if (this.state.selectedAirbnbs.includes(selectedAirbnb)) {
      alert('You already selected that listing.');
      return;
    }

    const newPayment = this.state.totalPayment + selectedAirbnb.payment.cost;

    this.setState(prevState => {
      return {
        selectedAirbnbs: [...prevState.selectedAirbnbs, selectedAirbnb],
        totalPayment: newPayment
      };
    });
  }

  deleteAirbnb = (index) => {
    this.setState(prevState => {
      const selectedAirbnb = this.state.selectedAirbnbs[index];
      const newSelectedAirbnbs = [...this.state.selectedAirbnbs];
      newSelectedAirbnbs.splice(index, 1);
      const newPayment = this.state.totalPayment - selectedAirbnb.payment.cost;

      return {
        selectedAirbnbs: newSelectedAirbnbs,
        totalPayment: newPayment
      };
    });
  }

  render() {
    return (
      <div className='App'>
        <div className='shoppingCartContainer'>
          <h2>Shopping Cart</h2>
          <ShoppingCart
            airbnbs={this.state.selectedAirbnbs}
            ondeleteAirbnb={this.deleteAirbnb}
            totalPayment={this.state.totalPayment}
          />
        </div>
        <hr />
        <div className='availablePropertiesContainer'>
          <h2>Available Properties</h2>
          <AirbnbList
            airbnbs={this.state.airbnbs}
            onbookAirbnb={this.bookAirbnb}
          />
        </div>
      </div>
    );
  }
}

export default App;
