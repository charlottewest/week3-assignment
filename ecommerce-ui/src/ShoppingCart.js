import React from 'react';
import PropTypes from 'prop-types';
import './ShoppingCart.css';

class ShoppingCart extends React.Component {
  static propTypes = {
    airbnbs: PropTypes.array,
    totalPayment: PropTypes.number
  }

  render() {
    const airbnbList = this.props.airbnbs
      .map((airbnb, index) => {
        return (
          <li key={index}>
            <span>{airbnb.title}</span>
            <button onClick={() => this.props.ondeleteAirbnb(index)}>Delete</button>
          </li>
        );
      });

    return (
      <div className='shoppingCart'>
        <ul>
          {airbnbList}
        </ul>
        <div className='totalPayment'>
          Total payment due: ${this.props.totalPayment}
        </div>
      </div>
    );
  }
}

export default ShoppingCart;
