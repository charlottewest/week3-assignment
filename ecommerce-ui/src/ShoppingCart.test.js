import React from 'react';
import ReactDOM from 'react-dom';
import {shallow, render, mount} from 'enzyme';
import App from './App';
import ShoppingCart from './ShoppingCart';
import Airbnbs from './airbnbs.json';

it('shallow renders', () => {
  const wrapper = shallow(<ShoppingCart airbnbs={Airbnbs}/>);
  expect(wrapper.render()).toMatchSnapshot();

  expect(wrapper.props('children')).toBeDefined();
});
