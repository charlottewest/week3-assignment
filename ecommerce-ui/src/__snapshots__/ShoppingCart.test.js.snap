// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<div
  class="shoppingCart"
>
  <ul>
    <li>
      <span>
        Centrally-located Manhattan studio
      </span>
      <button>
        Delete
      </button>
    </li>
    <li>
      <span>
        Denver apartment outside Union Station
      </span>
      <button>
        Delete
      </button>
    </li>
    <li>
      <span>
        Quaint 3-bedroom home in Gamla stan
      </span>
      <button>
        Delete
      </button>
    </li>
  </ul>
  <div
    class="totalPayment"
  >
    Total payment due: $
  </div>
</div>
`;
